pkgs/applications/emulators/cdemu/base.nix

```diff
@@ -1,12 +1,16 @@
-{ pname, version, pkgSha256 }:
-attrs@{ lib, stdenv, fetchurl, cmake, pkg-config, buildInputs, drvParams ? {}, ... }:
-stdenv.mkDerivation ( rec {
-  inherit pname version buildInputs;
+{ lib, stdenv, fetchurl, cmake, pkg-config
+, pname, version, sha256, buildInputs
+, nativeBuildInputs ? [ ]
+, postFixup ? ""
+, extraDrvParams ? { }
+}:
+stdenv.mkDerivation ( {
+  inherit pname version buildInputs postFixup;
   src = fetchurl {
     url = "mirror://sourceforge/cdemu/${pname}-${version}.tar.xz";
-    sha256 = pkgSha256;
+    inherit sha256;
   };
-  nativeBuildInputs = (attrs.nativeBuildInputs or [ ]) ++ [ pkg-config cmake ];
+  nativeBuildInputs = nativeBuildInputs ++ [ pkg-config cmake ];
   setSourceRoot = ''
```

Comment:
Why is `setSourceRoot` even needed? Also why not use `cmakeDir` and `cmakeFlags` and rely on the default `cmakeConfigurePhase`?

Reply:
I think cdemu might even predate the cmake build hook. In any case, it was my very first PR in this repo, so please forgive the clumsy style. I'll try your suggestion.

Reply:
yep, turns out that nowadays, the cmake hook does the right thing out of the box